### PR TITLE
chore: adding `MsgChannelUpgradeTimeout` fields and `sdk.Msg` interface impl

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

- Adding fields to MsgChannelUpgradeTimeout
- Implement ValidateBasic and GetSigners methods for sdk.Msg interface
- Adding tests

closes: #1626 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/core/04-channel/types/msgs.go [hunk 1]
```diff
@@ -571,18 +571,64 @@ var _ sdk.Msg = &MsgChannelUpgradeTimeout{}
...
+		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty proof")
+	}
+
+	if msg.ProofHeight.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "proof height must be non-zero")
+	}
+
+	if msg.CounterpartyChannel.State != OPEN {
+		return sdkerrors.Wrapf(ErrInvalidChannelState, "expected: %s, got: %s", OPEN, msg.CounterpartyChannel.State)
+	}
```
Comment (damiannolan) [inline modules/core/04-channel/types/msgs.go:614]: @AdityaSripal do you think there's any additional checks we need to perform on the counterparty channel? 
Maybe check the channel version is not empty?

File: modules/core/04-channel/types/msgs.go [hunk 2]
```diff
@@ -571,18 +571,64 @@ var _ sdk.Msg = &MsgChannelUpgradeTimeout{}
...
+
+	if len(msg.ProofChannel) == 0 {
+		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty proof")
+	}
+
+	if msg.ProofHeight.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "proof height must be non-zero")
+	}
+
+	if msg.CounterpartyChannel.State != OPEN {
```
Comment (charleenfei) [inline modules/core/04-channel/types/msgs.go:612]: could the counterparty channel state be in INITUPGRADE (crossing hellos?)

Comment (damiannolan) [reply modules/core/04-channel/types/msgs.go:612]: As far as I know crossing hellos are no longer supported, and won't be for channel upgrades? 
cc. @AdityaSripal

**Review by charleenfei — APPROVED** [review]
tentative approval if we don't support crossing hellos anymore

**Review by seantking — APPROVED** [review]

**Comment by damiannolan** [comment]
> tentative approval if we don't support crossing hellos anymore

I'm going to merge this. We can follow up with @AdityaSripal about this later 👍

**Pull request merged**